tractor/_actor.py

```diff
@@ -480,25 +501,30 @@ def _fork_main(
 
             # blocks here as expected until the channel server is
             # killed (i.e. this actor is cancelled or signalled by the parent)
-        except Exception:
+        except Exception as err:
+            if not registered_with_arbiter:
+                log.exception(
+                    f"Actor errored and failed to register with arbiter "
+                    f"@ {arbiter_addr}")
+
             if self._parent_chan:
                 try:
+                    # internal error so ship to parent without cid
                     await self._parent_chan.send(
-                        {'error': traceback.format_exc()})
+                        pack_error(err))
                 except trio.ClosedResourceError:
                     log.error(
                         f"Failed to ship error to parent "
                         f"{self._parent_chan.uid}, channel was closed")
                     log.exception("Actor errored:")
-
-            if not registered_with_arbiter:
-                log.exception(
-                    f"Actor errored and failed to register with arbiter "
-                    f"@ {arbiter_addr}")
             else:
+                # XXX wait, why?
+                # causes a hang if I always raise..
```

Comment:
grammar?

Reply:
I think it's ok. It's mostly for someone who knows the internals anyway.